src/components/dialog/dialog.ts

```diff
@@ -206,10 +206,10 @@ export default class IgcDialogComponent extends EventEmitterMixin<
     if (await this.toggleAnimation('close')) {
       this.open = false;
       this.animating = false;
-      this.emitEvent('igcClosed');
     }
 
     await this.updateComplete;
+    this.emitEvent('igcClosed');
```

Comment:
That `if (await this.toggleAnimation('close'))` is a bit odd, since the logic as it reads can potentially not close the dialog (set open to false), though that doesn't seem possible with the current implementation. After discussion, leaving for a separate PR.